ban/core/resource.py

```diff
@@ -126,25 +128,54 @@ def as_version(self):
         """Resources plus relations references and metadata."""
         return self.serialize({f: {} for f in self.versioned_fields})
 
+    @property
+    def status(self):
+        return 'deleted' if self.deleted_at else 'active'
+
+    @classmethod
+    def active(cls):
+        return cls.select().where(cls.deleted_at.is_null())
+
+    def mark_deleted(self):
+        if self.deleted_at:
+            raise ValueError('Resource already marked as deleted')
+        self.ensure_no_reverse_relation()
+        self.deleted_at = utcnow()
+        self.increment_version()
+        self.save()
+
+    def ensure_no_reverse_relation(self):
+        for name, field in self._meta.reverse_rel.items():
+            if getattr(self, name).count():
+                raise ValueError('Resource still linked by `{}`'.format(name))
```

Comment:
Now we are giving a message like "Resource still linked by `groups`", with only the first resource found.
You think it's worth going through all the linked resource to get the full list? It can be a lot.